events/events.go

```diff
@@ -31,9 +35,83 @@ type EventListener interface {
 	HandleEvent(event Event)
 }
 
+// EventEmitter describes an interface that can be used to emit events
+type EventEmitter interface {
+	EmitEvent(Event)
+}
+
 // EventRegistrar is an object that you can register EventListeners on.
 type EventRegistrar interface {
 	RegisterListener(EventListener)
+	DeregisterListener(EventListener)
```

Comment:
I would prefer `AddListener` and `RemoveListener` but since we have to be backwards compatible I guess you don't want to change the register method?